### PR TITLE
FEAT: Add Cart-management to the site

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

Cart management is to manage the cart of products that the user wants to buy. It contains all the products along with the price receipt.

### 🚀 New behavior

- Add products into the cart
- Remove products from the cart
- Increase the quantity of products in the cart.
- Price details are calculated accordingly based on price and discount.

### 🖼 Preview Link

[https://viskart-react-version-rfxd8mhmz-vishalpatil18.vercel.app/](https://viskart-react-version-rfxd8mhmz-vishalpatil18.vercel.app/)

### 💣 Is this a breaking change?
- [ ] Yes
- [x] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue #9 

> NOTE: You should login or signup in order to check its working. The final implementation would be done in the next PR.

### CODE REVIEW

File: src/components/FeaturedGames/FeaturedGames.jsx [hunk 1]
```diff
@@ -20,13 +20,7 @@ const FeaturedGames = ({ isHomePage = false }) => {
...
       <section className={styles.main_products}>
         {products.map((item) => (
-          <GamesCard
-            title={item.title}
-            vendor={item.vendor}
-            imageSrc={item.imageSrc}
-            price={item.price}
-            discount={item.discount}
-          />
+          <GamesCard key={item.id} item={item} />
```
Comment (kart-c) [inline src/components/FeaturedGames/FeaturedGames.jsx]: you are setting key twice here and in `GamesCard.jsx` as well. Also you are using `item.id` here and `item._id` in `GamesCard.jsx`. Can you check which one is correct?

File: src/services/removeFromCart.service.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const removeFromCartService = (item, cart, remove) => {
+  if (remove) {
+    return cart.filter((currItem) => currItem._id !== item._id);
+  }
+
+  return cart.reduce((acc, curr) => {
+    if (curr.id === item.id) {
+      if (item.quantity > 1)
+        acc = [...acc, { ...curr, quantity: curr.quantity - 1 }];
```
Comment (kart-c) [inline src/services/removeFromCart.service.js]: can you check if it is `id` or `_id`

**Review by kart-c — APPROVED** [review]
Code is well indented. Like the utilities function extracted out.

File: src/components/FeaturedGames/FeaturedGames.jsx (hunk 1 above)
Comment (VishalPatil18) [reply src/components/FeaturedGames/FeaturedGames.jsx]: Done 👍

File: src/services/removeFromCart.service.js (hunk 2 above)
Comment (VishalPatil18) [reply src/services/removeFromCart.service.js]: Done 👍

**Review by AnkurChunekar — APPROVED** [review]
Great work Vishal, code looks clean and functionalities work properly

**Review by AtulPant2704 — APPROVED** [review]
Everything looks good, note on the PR regarding login was also helpful.

**Pull request merged**